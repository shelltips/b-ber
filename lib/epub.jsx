
import cp from 'child_process'
import path from 'path'
import logger from './logger'
import conf from './config'

const exec = cp.exec
const timestamp = String(Date.now())
const bookname = `"${timestamp}.epub"`

const remove = 'epubs=`ls -1 *.epub 2>/dev/null | wc -l`; if [ $epubs != 0 ]; then rm *.epub; fi'
const compile = [
  `zip -X0 ${path.join(__dirname, '../', bookname)} ./mimetype`,
  `zip -X9Dr ${path.join(__dirname, '../', bookname)} ./META-INF -x *.DS_Store`,
  `zip -X9Dr ${path.join(__dirname, '../', bookname)} ./OPS -x *.DS_Store`
].join(' && ')

// TODO: assumes that epubcheck is symlinked to ./epubcheck
const validate = `java -jar epubcheck -e ${bookname}` // -e: only show fatal errors

const report = (err, stdout, stderr) => {
  const msg = []
  let error

  if (err) { throw err }
  if (stdout !== '') { msg.push(stdout) }
  if (stderr !== '') {
    msg.push(stderr)
    error = true
  }
  if (msg.length) { msg.map(_ => logger.info(_)) }
  if (error) { process.exit() }
}

const epub = () =>
  new Promise((resolve, reject) =>
    exec(remove, { cwd: './' }, (err1, stdout1, stderr1) => {
      report(err1, stdout1, stderr1)
      return exec(compile, { cwd: conf.dist }, (err2, stdout2, stderr2) => {
        report(err2, stdout2, stderr2)
        return exec(validate, { cwd: './' }, (err3, stdout3, stderr3) => {
          report(err3, stdout3, stderr3)
          resolve()
        })
      })
    })
  )

export default epub
