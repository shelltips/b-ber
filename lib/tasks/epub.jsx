
import zipper from 'epub-zipper'
import { log } from '../log'
import { dist } from '../utils'

const options = {
  input: dist(),
  output: process.cwd(),
  clean: true
}

const epub = () =>
  new Promise(resolve/* , reject */ =>
    zipper.create(options)
    .catch(err => log.error(err))
    .then(resolve)
  )

export default epub
