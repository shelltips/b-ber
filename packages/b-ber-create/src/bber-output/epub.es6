/**
 * @module epub
 */

import Promise from 'zousan'
import zipper from 'epub-zipper'
import log from '@canopycanopycanopy/b-ber-logger'
import { dist } from 'bber-utils'

/**
 * Compiles epub
 * @return {Promise<Object|Error>}
 */
const epub = () =>
    new Promise(resolve =>
        zipper.create({
            input: dist(),
            output: process.cwd(),
            clean: true,
        })
        .catch(err => log.error(err))
        .then(resolve)
    )


export default epub
