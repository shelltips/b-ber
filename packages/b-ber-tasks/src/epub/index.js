import zipper from 'epub-zipper'
import log from '@canopycanopycanopy/b-ber-logger'
import state from '@canopycanopycanopy/b-ber-lib/State'
import { getBookMetadata } from '@canopycanopycanopy/b-ber-lib/utils'

const epub = () =>
  process.argv.includes('--no-compile')
    ? Promise.resolve()
    : zipper
        .create({
          input: state.distDir,
          output: process.cwd(),
          clean: true,
          fileName: getBookMetadata('identifier', state),
        })
        .catch(log.error)

export default epub
