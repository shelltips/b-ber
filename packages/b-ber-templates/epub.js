'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _utils = require('@canopycanopycanopy/b-ber-lib/utils');

var epub = {
    portrait: function portrait(data) {
        return (!data.inline ? '<section epub:type="loi" title="Figures" class="chapter figures">' : '') + '\n            <div class="figure__large' + (data.inline ? ' figure__inline figure__inline--portrait ' + data.classes : '') + '"' + (0, _utils.getPagebreakAttribute)(data) + '>\n                <figure id="' + data.id + '">\n                    <div class="figure__items" style="width: 70%; margin: 0 auto;">\n                        ' + (!data.inline ? '<a href="' + data.ref + '.xhtml#ref' + data.id + '">' : '') + '\n                            <img class="portrait" alt="' + data.alt + '" src="../images/' + data.source + '" style="width: 100%; max-width: 100%; height: auto;"/>\n                        ' + (!data.inline ? '</a>' : '') + '\n                        <div class="figcaption" style="width: 100%; max-width: 100%; height: auto;">\n                            <p class="small">' + data.caption + '</p>\n                        </div>\n                    </div>\n                </figure>\n            </div>\n        ' + (!data.inline ? '</section>' : '');
    },
    landscape: function landscape(data) {
        return (!data.inline ? '<section epub:type="loi" title="Figures" class="chapter figures">' : '') + '\n            <div class="figure__large' + (data.inline ? ' figure__inline figure__inline--landscape ' + data.classes : '') + '"' + (0, _utils.getPagebreakAttribute)(data) + '>\n                <figure id="' + data.id + '">\n                    <div class="figure__items">\n                        ' + (!data.inline ? '<a href="' + data.ref + '.xhtml#ref' + data.id + '">' : '') + '\n                            <img class="landscape" alt="' + data.alt + '" src="../images/' + data.source + '" style="max-width: 100%;"/>\n                        ' + (!data.inline ? '</a>' : '') + '\n                        <div class="figcaption" style="max-width: 100%;">\n                            <p class="small">' + data.caption + '</p>\n                        </div>\n                    </div>\n                </figure>\n            </div>\n        ' + (!data.inline ? '</section>' : '');
    },

    'portrait-high': function portraitHigh(data) {
        // eslint-disable-line func-names
        return (!data.inline ? '<section epub:type="loi" title="Figures" class="chapter figures">' : '') + '\n            <div class="figure__large' + (data.inline ? ' figure__inline figure__inline--portrait-high ' + data.classes : '') + '"' + (0, _utils.getPagebreakAttribute)(data) + '>\n                <figure id="' + data.id + '">\n                    <div class="figure__items" style="width: 60%; margin: 0 auto;">\n                        ' + (!data.inline ? '<a href="' + data.ref + '.xhtml#ref' + data.id + '">' : '') + '\n                            <img class="portrait-long" alt="' + data.alt + '" src="../images/' + data.source + '" style="width: 100%; max-width: 100%; height: auto;"/>\n                        ' + (!data.inline ? '</a>' : '') + '\n                        <div class="figcaption" style="width: 100%; max-width: 100%; height: auto;">\n                            <p class="small">' + data.caption + '</p>\n                        </div>\n                    </div>\n                </figure>\n            </div>\n        ' + (!data.inline ? '</section>' : '');
    },
    square: function square(data) {
        return (!data.inline ? '<section epub:type="loi" title="Figures" class="chapter figures">' : '') + '\n            <div class="figure__large' + (data.inline ? ' figure__inline figure__inline--square ' + data.classes : '') + '"' + (0, _utils.getPagebreakAttribute)(data) + '>\n                <figure id="' + data.id + '">\n                    <div class="figure__items" style="width: 85%; margin: 0 auto;">\n                        ' + (!data.inline ? '<a href="' + data.ref + '.xhtml#ref' + data.id + '">' : '') + '\n                            <img class="square" alt="' + data.alt + '" src="../images/' + data.source + '" style="width: 100%; max-width: 100%; height: auto;"/>\n                        ' + (!data.inline ? '</a>' : '') + '\n                        <div class="figcaption" style="width: 100%; max-width: 100%; height: auto;">\n                            <p class="small">' + data.caption + '</p>\n                        </div>\n                    </div>\n                </figure>\n            </div>\n        ' + (!data.inline ? '</section>' : '');
    },
    audio: function audio(data) {
        return (!data.inline ? '<section epub:type="loi" title="Figures" class="chapter figures">' : '') + '\n            <div class="figure__large">\n                <figure id="' + data.id + '">\n\n                    <div class="figure__items">\n\n                        <div class="audio">\n                            <audio ' + data.attrString + '>\n                                ' + data.sourceElements + '\n                                <div class="media__fallback__' + data.mediaType + ' media__fallback--image figure__small--landscape figure__small">\n                                    <figure>\n                                        <img src="' + data.poster + '" alt="Media fallback image"/>\n                                    </figure>\n                                </div>\n                                <p class="media__fallback__' + data.mediaType + ' media__fallback--text">Your device does not support the HTML5 ' + data.mediaType + ' API.</p>\n                            </audio>\n                        </div>\n\n                        <div class="figcaption" style="max-width: 100%;">\n                            <p class="small">\n                                ' + (data.caption ? data.caption + '<br/>' : '') + '\n                                <a href="' + data.ref + '.xhtml#ref' + data.id + '">Return</a>\n                            </p>\n                        </div>\n\n                    </div>\n\n                </figure>\n            </div>\n        ' + (!data.inline ? '</section>' : '');
    },
    video: function video(data) {
        return (!data.inline ? '<section epub:type="loi" title="Figures" class="chapter figures">' : '') + '\n            <div class="figure__large">\n                <figure id="' + data.id + '">\n\n                    <div class="figure__items">\n\n                        <div class="video">\n                            <video ' + data.attrString + '>\n                                ' + data.sourceElements + '\n                                <div class="media__fallback__' + data.mediaType + ' media__fallback--image figure__small--landscape figure__small">\n                                    <figure>\n                                        <img src="' + data.poster + '" alt="Media fallback image"/>\n                                    </figure>\n                                </div>\n                                <p class="media__fallback__' + data.mediaType + ' media__fallback--text">Your device does not support the HTML5 ' + data.mediaType + ' API.</p>\n                            </video>\n                        </div>\n\n                        <div class="figcaption" style="max-width: 100%;">\n                            <p class="small">\n                                ' + (data.caption ? data.caption + '<br/>' : '') + '\n                                <a href="' + data.ref + '.xhtml#ref' + data.id + '">Return</a>\n                            </p>\n                        </div>\n\n                    </div>\n\n                </figure>\n            </div>\n        ' + (!data.inline ? '</section>' : '');
    },
    iframe: function iframe(data) {
        return (!data.inline ? '<section epub:type="loi" title="Figures" class="chapter figures">' : '') + '\n            <div class="figure__large">\n                <figure id="' + data.id + '">\n                    <div class="figure__items">\n                        <div class="iframe">\n                            <iframe src="' + (0, _utils.encodeQueryString)(data.source) + '" />\n                        </div>\n\n                        <div class="figcaption" style="max-width: 100%;">\n                            <p class="small">\n                                ' + (data.caption ? data.caption + '<br/>' : '') + '\n                                <a href="' + data.ref + '.xhtml#ref' + data.id + '">Return</a>\n                            </p>\n                        </div>\n                    </div>\n                </figure>\n            </div>\n        ' + (!data.inline ? '</section>' : '');
    }
}; /* eslint-disable max-len */

exports.default = epub;