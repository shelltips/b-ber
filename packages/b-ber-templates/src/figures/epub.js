import { media, iframe, figure } from './helpers'

const epub = {
  portrait: data => figure({ data, ratioName: 'portrait' }),
  landscape: data => figure({ data, ratioName: 'landscape' }),
  'portrait-high': data => figure({ data, ratioName: 'portrait-high' }),
  square: data => figure({ data, ratioName: 'square' }),
  audio: data => media({ ...data, applyInlineClasses: true }),
  video: data => media({ ...data, applyInlineClasses: true }),
  iframe: data => iframe({ ...data, applyInlineClasses: true }),
}

export default epub
