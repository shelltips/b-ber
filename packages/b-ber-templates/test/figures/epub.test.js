import epub from '../../src/figures/epub'

describe('templates.epub', () => {
    test('creates the markup', () => {
        const data = {
            inline: true,
            classes: ['test'],
            pagebreak: '',
            id: 1,
            ref: 1,
            alt: 'test',
            source: 'test',
            caption: 'test',
        }

        expect(epub.portrait(data)).toMatchSnapshot()
        expect(epub.landscape(data)).toMatchSnapshot()
        expect(epub['portrait-high'](data)).toMatchSnapshot()
        expect(epub.square(data)).toMatchSnapshot()
        expect(epub.audio(data)).toMatchSnapshot()
        expect(epub.video(data)).toMatchSnapshot()
        expect(epub.iframe(data)).toMatchSnapshot()

        data.inline = false
        expect(epub.portrait(data)).toMatchSnapshot()
        expect(epub.landscape(data)).toMatchSnapshot()
        expect(epub['portrait-high'](data)).toMatchSnapshot()
        expect(epub.square(data)).toMatchSnapshot()
        expect(epub.audio(data)).toMatchSnapshot()
        expect(epub.video(data)).toMatchSnapshot()
        expect(epub.iframe(data)).toMatchSnapshot()

        data.pagebreak = 'before'
        expect(epub.portrait(data)).toMatchSnapshot()
        expect(epub.landscape(data)).toMatchSnapshot()
        expect(epub['portrait-high'](data)).toMatchSnapshot()
        expect(epub.square(data)).toMatchSnapshot()
        expect(epub.audio(data)).toMatchSnapshot()
        expect(epub.video(data)).toMatchSnapshot()
        expect(epub.iframe(data)).toMatchSnapshot()

        data.pagebreak = 'after'
        expect(epub.portrait(data)).toMatchSnapshot()
        expect(epub.landscape(data)).toMatchSnapshot()
        expect(epub['portrait-high'](data)).toMatchSnapshot()
        expect(epub.square(data)).toMatchSnapshot()
        expect(epub.audio(data)).toMatchSnapshot()
        expect(epub.video(data)).toMatchSnapshot()
        expect(epub.iframe(data)).toMatchSnapshot()


    })
})
