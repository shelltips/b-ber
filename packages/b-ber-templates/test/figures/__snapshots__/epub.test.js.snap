// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates.epub creates the markup 1`] = `
"
    
        <div class=\\"figure__large figure__inline figure__inline--portrait test\\" >
            <figure id=\\"1\\">
                <div class=\\"figure__items\\" style=\\"width:70%;margin:0 auto;\\">
                    
                        <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test\\" style=\\"width:100%;max-width:100%;height:auto;\\"/>
                    
                    <div class=\\"figcaption\\" style=\\"width:100%;max-width:100%;height:auto;\\">
                        <p class=\\"small\\">test</p>
                    </div>
                </div>
            </figure>
        </div>
    
"
`;

exports[`templates.epub creates the markup 2`] = `
"
    
        <div class=\\"figure__large figure__inline figure__inline--landscape test\\" >
            <figure id=\\"1\\">
                <div class=\\"figure__items\\" >
                    
                        <img class=\\"landscape\\" alt=\\"test\\" src=\\"../images/test\\" style=\\"max-width:100%;\\"/>
                    
                    <div class=\\"figcaption\\" style=\\"max-width:100%;\\">
                        <p class=\\"small\\">test</p>
                    </div>
                </div>
            </figure>
        </div>
    
"
`;

exports[`templates.epub creates the markup 3`] = `
"
    
        <div class=\\"figure__large figure__inline figure__inline--portrait-high test\\" >
            <figure id=\\"1\\">
                <div class=\\"figure__items\\" style=\\"width:60%;margin:0 auto;\\">
                    
                        <img class=\\"portrait-high\\" alt=\\"test\\" src=\\"../images/test\\" style=\\"width:100%;max-width:100%;height:auto;\\"/>
                    
                    <div class=\\"figcaption\\" style=\\"width:100%;max-width:100%;height:auto;\\">
                        <p class=\\"small\\">test</p>
                    </div>
                </div>
            </figure>
        </div>
    
"
`;

exports[`templates.epub creates the markup 4`] = `
"
    
        <div class=\\"figure__large figure__inline figure__inline--square test\\" >
            <figure id=\\"1\\">
                <div class=\\"figure__items\\" style=\\"width:85%;margin:0 auto;\\">
                    
                        <img class=\\"square\\" alt=\\"test\\" src=\\"../images/test\\" style=\\"width:100%;max-width:100%;height:auto;\\"/>
                    
                    <div class=\\"figcaption\\" style=\\"width:100%;max-width:100%;height:auto;\\">
                        <p class=\\"small\\">test</p>
                    </div>
                </div>
            </figure>
        </div>
    
"
`;

exports[`templates.epub creates the markup 5`] = `
"
    
        <div class=\\"figure__large figure__inline--square\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\">
                    <div class=\\"audio-or-video\\">
                        <audio-or-video autoplay=\\"false\\">
                            <source />
                            <div class=\\"media__fallback__audio-or-video media__fallback--image figure__small--landscape figure__small\\">
                                <figure>
                                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                                </figure>
                            </div>
                            <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
                        </audio-or-video>
                    </div>
                    <div class=\\"figcaption\\" style=\\"max-width: 100%;\\">
                        <p class=\\"small\\">
                            test<br/>
                            <a href=\\"1.xhtml#ref1\\">Return</a>
                        </p>
                    </div>
                </div>
            </figure>
        </div>
    
    "
`;

exports[`templates.epub creates the markup 6`] = `
"
    
        <div class=\\"figure__large figure__inline--square\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\">
                    <div class=\\"audio-or-video\\">
                        <audio-or-video autoplay=\\"false\\">
                            <source />
                            <div class=\\"media__fallback__audio-or-video media__fallback--image figure__small--landscape figure__small\\">
                                <figure>
                                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                                </figure>
                            </div>
                            <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
                        </audio-or-video>
                    </div>
                    <div class=\\"figcaption\\" style=\\"max-width: 100%;\\">
                        <p class=\\"small\\">
                            test<br/>
                            <a href=\\"1.xhtml#ref1\\">Return</a>
                        </p>
                    </div>
                </div>
            </figure>
        </div>
    
    "
`;

exports[`templates.epub creates the markup 7`] = `
"
    
        <div class=\\"figure__large\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\">
                    <div class=\\"iframe\\">
                        <iframe src=\\"test\\" />
                    </div>

                    <div class=\\"figcaption\\" style=\\"max-width: 100%;\\">
                        <p class=\\"small\\">
                            test<br/>
                            <a href=\\"1.xhtml#ref1\\">Return</a>
                        </p>
                    </div>
                </div>
            </figure>
        </div>
    
    "
`;

exports[`templates.epub creates the markup 8`] = `
"
    <section epub:type=\\"loi\\" title=\\"Figures\\" class=\\"chapter figures\\">
        <div class=\\"figure__large\\" >
            <figure id=\\"1\\">
                <div class=\\"figure__items\\" style=\\"width:70%;margin:0 auto;\\">
                    <a href=\\"1.xhtml#ref1\\">
                        <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test\\" style=\\"width:100%;max-width:100%;height:auto;\\"/>
                    </a>
                    <div class=\\"figcaption\\" style=\\"width:100%;max-width:100%;height:auto;\\">
                        <p class=\\"small\\">test</p>
                    </div>
                </div>
            </figure>
        </div>
    </section>
"
`;

exports[`templates.epub creates the markup 9`] = `
"
    <section epub:type=\\"loi\\" title=\\"Figures\\" class=\\"chapter figures\\">
        <div class=\\"figure__large\\" >
            <figure id=\\"1\\">
                <div class=\\"figure__items\\" >
                    <a href=\\"1.xhtml#ref1\\">
                        <img class=\\"landscape\\" alt=\\"test\\" src=\\"../images/test\\" style=\\"max-width:100%;\\"/>
                    </a>
                    <div class=\\"figcaption\\" style=\\"max-width:100%;\\">
                        <p class=\\"small\\">test</p>
                    </div>
                </div>
            </figure>
        </div>
    </section>
"
`;

exports[`templates.epub creates the markup 10`] = `
"
    <section epub:type=\\"loi\\" title=\\"Figures\\" class=\\"chapter figures\\">
        <div class=\\"figure__large\\" >
            <figure id=\\"1\\">
                <div class=\\"figure__items\\" style=\\"width:60%;margin:0 auto;\\">
                    <a href=\\"1.xhtml#ref1\\">
                        <img class=\\"portrait-high\\" alt=\\"test\\" src=\\"../images/test\\" style=\\"width:100%;max-width:100%;height:auto;\\"/>
                    </a>
                    <div class=\\"figcaption\\" style=\\"width:100%;max-width:100%;height:auto;\\">
                        <p class=\\"small\\">test</p>
                    </div>
                </div>
            </figure>
        </div>
    </section>
"
`;

exports[`templates.epub creates the markup 11`] = `
"
    <section epub:type=\\"loi\\" title=\\"Figures\\" class=\\"chapter figures\\">
        <div class=\\"figure__large\\" >
            <figure id=\\"1\\">
                <div class=\\"figure__items\\" style=\\"width:85%;margin:0 auto;\\">
                    <a href=\\"1.xhtml#ref1\\">
                        <img class=\\"square\\" alt=\\"test\\" src=\\"../images/test\\" style=\\"width:100%;max-width:100%;height:auto;\\"/>
                    </a>
                    <div class=\\"figcaption\\" style=\\"width:100%;max-width:100%;height:auto;\\">
                        <p class=\\"small\\">test</p>
                    </div>
                </div>
            </figure>
        </div>
    </section>
"
`;

exports[`templates.epub creates the markup 12`] = `
"
    <section epub:type=\\"loi\\" title=\\"Figures\\" class=\\"chapter figures\\">
        <div class=\\"figure__large\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\">
                    <div class=\\"audio-or-video\\">
                        <audio-or-video autoplay=\\"false\\">
                            <source />
                            <div class=\\"media__fallback__audio-or-video media__fallback--image figure__small--landscape figure__small\\">
                                <figure>
                                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                                </figure>
                            </div>
                            <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
                        </audio-or-video>
                    </div>
                    <div class=\\"figcaption\\" style=\\"max-width: 100%;\\">
                        <p class=\\"small\\">
                            test<br/>
                            <a href=\\"1.xhtml#ref1\\">Return</a>
                        </p>
                    </div>
                </div>
            </figure>
        </div>
    </section>
    "
`;

exports[`templates.epub creates the markup 13`] = `
"
    <section epub:type=\\"loi\\" title=\\"Figures\\" class=\\"chapter figures\\">
        <div class=\\"figure__large figure__inline--square\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\">
                    <div class=\\"audio-or-video\\">
                        <audio-or-video autoplay=\\"false\\">
                            <source />
                            <div class=\\"media__fallback__audio-or-video media__fallback--image figure__small--landscape figure__small\\">
                                <figure>
                                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                                </figure>
                            </div>
                            <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
                        </audio-or-video>
                    </div>
                    <div class=\\"figcaption\\" style=\\"max-width: 100%;\\">
                        <p class=\\"small\\">
                            test<br/>
                            <a href=\\"1.xhtml#ref1\\">Return</a>
                        </p>
                    </div>
                </div>
            </figure>
        </div>
    </section>
    "
`;

exports[`templates.epub creates the markup 14`] = `
"
    <section epub:type=\\"loi\\" title=\\"Figures\\" class=\\"chapter figures\\">
        <div class=\\"figure__large\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\">
                    <div class=\\"iframe\\">
                        <iframe src=\\"test\\" />
                    </div>

                    <div class=\\"figcaption\\" style=\\"max-width: 100%;\\">
                        <p class=\\"small\\">
                            test<br/>
                            <a href=\\"1.xhtml#ref1\\">Return</a>
                        </p>
                    </div>
                </div>
            </figure>
        </div>
    </section>
    "
`;

exports[`templates.epub creates the markup 15`] = `
"
    
        <div class=\\"figure__large figure__inline figure__inline--portrait test\\"  style=\\"page-break-before:always;\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\" style=\\"width:70%;margin:0 auto;\\">
                    
                        <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test\\" style=\\"width:100%;max-width:100%;height:auto;\\"/>
                    
                    <div class=\\"figcaption\\" style=\\"width:100%;max-width:100%;height:auto;\\">
                        <p class=\\"small\\">test</p>
                    </div>
                </div>
            </figure>
        </div>
    
"
`;

exports[`templates.epub creates the markup 16`] = `
"
    
        <div class=\\"figure__large figure__inline figure__inline--landscape test\\"  style=\\"page-break-before:always;\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\" >
                    
                        <img class=\\"landscape\\" alt=\\"test\\" src=\\"../images/test\\" style=\\"max-width:100%;\\"/>
                    
                    <div class=\\"figcaption\\" style=\\"max-width:100%;\\">
                        <p class=\\"small\\">test</p>
                    </div>
                </div>
            </figure>
        </div>
    
"
`;

exports[`templates.epub creates the markup 17`] = `
"
    
        <div class=\\"figure__large figure__inline figure__inline--portrait-high test\\"  style=\\"page-break-before:always;\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\" style=\\"width:60%;margin:0 auto;\\">
                    
                        <img class=\\"portrait-high\\" alt=\\"test\\" src=\\"../images/test\\" style=\\"width:100%;max-width:100%;height:auto;\\"/>
                    
                    <div class=\\"figcaption\\" style=\\"width:100%;max-width:100%;height:auto;\\">
                        <p class=\\"small\\">test</p>
                    </div>
                </div>
            </figure>
        </div>
    
"
`;

exports[`templates.epub creates the markup 18`] = `
"
    
        <div class=\\"figure__large figure__inline figure__inline--square test\\"  style=\\"page-break-before:always;\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\" style=\\"width:85%;margin:0 auto;\\">
                    
                        <img class=\\"square\\" alt=\\"test\\" src=\\"../images/test\\" style=\\"width:100%;max-width:100%;height:auto;\\"/>
                    
                    <div class=\\"figcaption\\" style=\\"width:100%;max-width:100%;height:auto;\\">
                        <p class=\\"small\\">test</p>
                    </div>
                </div>
            </figure>
        </div>
    
"
`;

exports[`templates.epub creates the markup 19`] = `
"
    
        <div class=\\"figure__large figure__inline--square\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\">
                    <div class=\\"audio-or-video\\">
                        <audio-or-video autoplay=\\"false\\">
                            <source />
                            <div class=\\"media__fallback__audio-or-video media__fallback--image figure__small--landscape figure__small\\">
                                <figure>
                                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                                </figure>
                            </div>
                            <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
                        </audio-or-video>
                    </div>
                    <div class=\\"figcaption\\" style=\\"max-width: 100%;\\">
                        <p class=\\"small\\">
                            test<br/>
                            <a href=\\"1.xhtml#ref1\\">Return</a>
                        </p>
                    </div>
                </div>
            </figure>
        </div>
    
    "
`;

exports[`templates.epub creates the markup 20`] = `
"
    
        <div class=\\"figure__large figure__inline--square\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\">
                    <div class=\\"audio-or-video\\">
                        <audio-or-video autoplay=\\"false\\">
                            <source />
                            <div class=\\"media__fallback__audio-or-video media__fallback--image figure__small--landscape figure__small\\">
                                <figure>
                                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                                </figure>
                            </div>
                            <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
                        </audio-or-video>
                    </div>
                    <div class=\\"figcaption\\" style=\\"max-width: 100%;\\">
                        <p class=\\"small\\">
                            test<br/>
                            <a href=\\"1.xhtml#ref1\\">Return</a>
                        </p>
                    </div>
                </div>
            </figure>
        </div>
    
    "
`;

exports[`templates.epub creates the markup 21`] = `
"
    
        <div class=\\"figure__large\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\">
                    <div class=\\"iframe\\">
                        <iframe src=\\"test\\" />
                    </div>

                    <div class=\\"figcaption\\" style=\\"max-width: 100%;\\">
                        <p class=\\"small\\">
                            test<br/>
                            <a href=\\"1.xhtml#ref1\\">Return</a>
                        </p>
                    </div>
                </div>
            </figure>
        </div>
    
    "
`;

exports[`templates.epub creates the markup 22`] = `
"
    
        <div class=\\"figure__large figure__inline figure__inline--portrait test\\"  style=\\"page-break-after:always;\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\" style=\\"width:70%;margin:0 auto;\\">
                    
                        <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test\\" style=\\"width:100%;max-width:100%;height:auto;\\"/>
                    
                    <div class=\\"figcaption\\" style=\\"width:100%;max-width:100%;height:auto;\\">
                        <p class=\\"small\\">test</p>
                    </div>
                </div>
            </figure>
        </div>
    
"
`;

exports[`templates.epub creates the markup 23`] = `
"
    
        <div class=\\"figure__large figure__inline figure__inline--landscape test\\"  style=\\"page-break-after:always;\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\" >
                    
                        <img class=\\"landscape\\" alt=\\"test\\" src=\\"../images/test\\" style=\\"max-width:100%;\\"/>
                    
                    <div class=\\"figcaption\\" style=\\"max-width:100%;\\">
                        <p class=\\"small\\">test</p>
                    </div>
                </div>
            </figure>
        </div>
    
"
`;

exports[`templates.epub creates the markup 24`] = `
"
    
        <div class=\\"figure__large figure__inline figure__inline--portrait-high test\\"  style=\\"page-break-after:always;\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\" style=\\"width:60%;margin:0 auto;\\">
                    
                        <img class=\\"portrait-high\\" alt=\\"test\\" src=\\"../images/test\\" style=\\"width:100%;max-width:100%;height:auto;\\"/>
                    
                    <div class=\\"figcaption\\" style=\\"width:100%;max-width:100%;height:auto;\\">
                        <p class=\\"small\\">test</p>
                    </div>
                </div>
            </figure>
        </div>
    
"
`;

exports[`templates.epub creates the markup 25`] = `
"
    
        <div class=\\"figure__large figure__inline figure__inline--square test\\"  style=\\"page-break-after:always;\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\" style=\\"width:85%;margin:0 auto;\\">
                    
                        <img class=\\"square\\" alt=\\"test\\" src=\\"../images/test\\" style=\\"width:100%;max-width:100%;height:auto;\\"/>
                    
                    <div class=\\"figcaption\\" style=\\"width:100%;max-width:100%;height:auto;\\">
                        <p class=\\"small\\">test</p>
                    </div>
                </div>
            </figure>
        </div>
    
"
`;

exports[`templates.epub creates the markup 26`] = `
"
    
        <div class=\\"figure__large figure__inline--square\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\">
                    <div class=\\"audio-or-video\\">
                        <audio-or-video autoplay=\\"false\\">
                            <source />
                            <div class=\\"media__fallback__audio-or-video media__fallback--image figure__small--landscape figure__small\\">
                                <figure>
                                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                                </figure>
                            </div>
                            <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
                        </audio-or-video>
                    </div>
                    <div class=\\"figcaption\\" style=\\"max-width: 100%;\\">
                        <p class=\\"small\\">
                            test<br/>
                            <a href=\\"1.xhtml#ref1\\">Return</a>
                        </p>
                    </div>
                </div>
            </figure>
        </div>
    
    "
`;

exports[`templates.epub creates the markup 27`] = `
"
    
        <div class=\\"figure__large figure__inline--square\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\">
                    <div class=\\"audio-or-video\\">
                        <audio-or-video autoplay=\\"false\\">
                            <source />
                            <div class=\\"media__fallback__audio-or-video media__fallback--image figure__small--landscape figure__small\\">
                                <figure>
                                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                                </figure>
                            </div>
                            <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
                        </audio-or-video>
                    </div>
                    <div class=\\"figcaption\\" style=\\"max-width: 100%;\\">
                        <p class=\\"small\\">
                            test<br/>
                            <a href=\\"1.xhtml#ref1\\">Return</a>
                        </p>
                    </div>
                </div>
            </figure>
        </div>
    
    "
`;

exports[`templates.epub creates the markup 28`] = `
"
    
        <div class=\\"figure__large\\">
            <figure id=\\"1\\">
                <div class=\\"figure__items\\">
                    <div class=\\"iframe\\">
                        <iframe src=\\"test\\" />
                    </div>

                    <div class=\\"figcaption\\" style=\\"max-width: 100%;\\">
                        <p class=\\"small\\">
                            test<br/>
                            <a href=\\"1.xhtml#ref1\\">Return</a>
                        </p>
                    </div>
                </div>
            </figure>
        </div>
    
    "
`;
